src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExercise.java

```diff
@@ -811,4 +821,29 @@ public Set<ExerciseHint> getExerciseHints() {
     public void setExerciseHints(Set<ExerciseHint> exerciseHints) {
         this.exerciseHints = exerciseHints;
     }
+
+    public void setBuildPlan(BuildPlan buildPlan) {
+        this.buildPlan = buildPlan;
+    }
+
+    // TODO: Safe build plan when exercise is created
+    public BuildPlan getBuildPlan() {
+        return buildPlan;
+    }
+
+    public void setBuildPlanAccessSecret(String buildPlanAccessSecret) {
```

Comment:
Now that a `BuildPlan` class exists, this should probably be an Attribute `accessSecret` in `BuildPlan`.

Reply:
It is planned to make it possible to change the build plan throughout the exercise lifecycle.
When setting up the repository in GitLab, the URL to the build plan and the accessSecret is configured in GitLab (e.g. `http://artemis/api/programming-exercises/1/build-plan?secret=abced&file-extension=.yml`).
This URL should still be used, even if the build plan of the exercise is changed. Thus, no additional API calls have to be made to update the secret in every repository.